conda/install.py

```diff
@@ -922,6 +927,8 @@ def move_path_to_trash(path):
         except OSError as e:
             log.debug("Could not move %s to %s (%s)" % (path, trash_dir, e))
         else:
+            if path in linked_data_:
+                del linked_data_[path]
```

Comment:
> path can be anything, but it only deletes it when it's found as a key in linked_data_ which will correspond to path being a prefix.

Sure, I understand that, but the problem is that if it matches a _directory_ inside the prefix, it will fail to remove the dirty data.